nemo_skills/pipeline/utils.py

```diff
@@ -266,6 +267,100 @@ def get_packager():
         )
 
 
+def get_env_variables(cluster_config):
+    """
+    Will get the environment variables from the cluster config and the user environment.
+
+    The following items in the cluster config are supported:
+    - `required_env_vars` - list of required environment variables
+    - `env_vars` - list of optional environment variables
+
+    Args:
+        cluster_config: cluster config dictionary
+
+    Returns:
+        dict: dictionary of environment
+    """
+    env_vars = {}
+    # Check for user requested env variables
+    required_env_vars = cluster_config.get("required_env_vars", [])
+    for env_var in required_env_vars:
+        if env_var not in os.environ:
+            raise ValueError(f"Required environment variable {env_var} not found.")
+        env_vars[env_var] = os.environ[env_var]
+
+    # Add optional env variables
+    optional_env_vars = cluster_config.get("env_vars", [])
+    for env_var in optional_env_vars:
+        if env_var in os.environ:
+            logging.info(f"Adding optional environment variable {env_var} (value={os.environ[env_var]})")
+            env_vars[env_var] = os.environ[env_var]
+        else:
+            logging.info(f"Optional environment variable {env_var} not found in user environment; skipping.")
+
+    return env_vars
+
+
+def get_mounts_from_config(cluster_config: dict, env_vars: dict = None):
+    """
+    Determines if there are mount paths that are being passed via environment variables.
+    Selects the special key in the cluster config called `env_mounts` which is a list of strings.
+    Each string is in the format of `env_var:mount_path` where `env_var` is the name of the environment variable
+    and `mount_path` is the path where the environment variable is mounted.
+
+    Args:
+        cluster_config (dict): cluster config dictionary
+        env_vars (dict): dictionary of environment variables
+
+    Returns:
+        list: updated list of mounts
+    """
+    mounts = cluster_config.get('mounts', [])
+
+    # if env_vars is None, we will get the env_vars from the cluster config
+    if env_vars is None:
+        env_vars = get_env_variables(cluster_config)
+
+    # if there are no env_mounts, we will return the mounts as is
+    if 'env_mounts' not in cluster_config:
+        return mounts
+
+    # if there are env_mounts, we will add the mounts from the env_mounts
+    env_mounts = cluster_config['env_mounts']
```

Comment:
I think it's better to just support it in the mounts field. So we'd only have a single mounts field that can be either normal mounts or from env vars like in your current logic